core/cmd/local_client.go

```diff
@@ -83,19 +84,12 @@ func applicationLockDB(cfg config.GeneralConfig, db *sqlx.DB, lggr logger.Logger
 	case "advisorylock", "dual":
 		advisoryLock = pg.NewAdvisoryLock(db, cfg.AdvisoryLockID(), lggr, cfg.AdvisoryLockCheckInterval())
 		if err = advisoryLock.TakeAndHold(); err != nil {
-			return cleanup, errors.Wrap(err, "error acquiring lock")
+			return nil, errors.Wrap(err, "error acquiring lock")
 		}
+		release = advisoryLock.Release
 	}
 
-	cleanup = func() {
-		if leaseLock != nil {
-			leaseLock.Release()
-		}
-		if advisoryLock != nil {
-			advisoryLock.Release()
-		}
-	}
-	return cleanup, nil
+	return release, nil
```

Comment:
It looks like this was split brained before, but I think this latter part was correct to check both locks so that `dual` is released properly.